src/category_theory/equivalence.lean

```diff
@@ -196,20 +196,14 @@ omit 𝒟 ℰ
 -- The power structure is nevertheless useful
 
 /-- Powers of an auto-equivalence. -/
-def nat_pow (e : C ≌ C) : ℕ → (C ≌ C)
-| 0 := equivalence.refl
-| 1 := e
-| (n+2) := e.trans (nat_pow (n+1))
-
-instance has_pow_nat : has_pow (C ≌ C) ℕ := ⟨nat_pow⟩
-
-/-- Integer powers of an auto-equivalence. -/
-def int_pow (e : C ≌ C) : ℤ → (C ≌ C)
-| (int.of_nat n) := e^n
+def pow (e : C ≌ C) : ℤ → (C ≌ C)
```

Comment:
Aren't equivalences a `group` already? Then this should come for free. There is also the `iterate` API, that might be useful here.

Reply:
I don't think they are a group yet. For endomorphisms `X ⟶ X` and automorphisms `X \iso X`, we don't directly provide monoid and group instances, because notationally it would be too confusing, but instead provide type synonyms `End X` and `Aut X`. We could do that for `Endofunctor C` and `Autoequiv C`, but it doesn't exist yet. Of course, these are only barely monoids and groups: really they are higher categorical things, with natural isomorphisms for identitors and associators.

Reply:
Ok, fair enough.